src/useAccessibilityInfo.ts

```diff
@@ -1,43 +1,94 @@
+  AccessibilityInfo.isGrayscaleEnabled &&
+  AccessibilityInfo.isInvertColorsEnabled &&
+  AccessibilityInfo.isReduceMotionEnabled &&
+  AccessibilityInfo.isReduceTransparencyEnabled
+)
 
-  const handleReduceMotionChanged = (enabled: AccessibilityChangeEvent) =>
-    setReduceMotionEnabled(enabled)
-  const handleScreenReaderChanged = (enabled: AccessibilityChangeEvent) =>
-    setScreenReaderEnabled(enabled)
+type AccessibilityInfoStaticInitializers =
+  | 'isBoldTextEnabled'
+  | 'isScreenReaderEnabled'
+  | 'isGrayscaleEnabled'
+  | 'isInvertColorsEnabled'
+  | 'isReduceMotionEnabled'
+  | 'isReduceTransparencyEnabled'
+
+type AccessibilityEventToInfoStaticKeyMap = {
+  [K in AccessibilityChangeEventName]?: AccessibilityInfoStaticInitializers
+}
+
+const EVENT_NAME_TO_INITIALIZER: AccessibilityEventToInfoStaticKeyMap = {
+  boldTextChanged: 'isBoldTextEnabled',
+  screenReaderChanged: 'isScreenReaderEnabled',
+  grayscaleChanged: 'isGrayscaleEnabled',
+  invertColorsChanged: 'isInvertColorsEnabled',
+  reduceMotionChanged: 'isReduceMotionEnabled',
+  reduceTransparencyChanged: 'isReduceTransparencyEnabled',
+}
+
+function useAccessibilityStateListener(
+  eventName: AccessibilityChangeEventName,
+): boolean {
+  const [isEnabled, setIsEnabled] = useState(false)
 
   useEffect(() => {
-    AccessibilityInfo.isReduceMotionEnabled().then(handleReduceMotionChanged)
-    AccessibilityInfo.isScreenReaderEnabled().then(handleScreenReaderChanged)
-
-    AccessibilityInfo.addEventListener(
-      'reduceMotionChanged',
-      handleReduceMotionChanged as (event: AccessibilityEvent) => void,
-    )
-    AccessibilityInfo.addEventListener(
-      'screenReaderChanged',
-      handleScreenReaderChanged as (event: AccessibilityEvent) => void,
-    )
-
-    return () => {
-      AccessibilityInfo.removeEventListener(
-        'reduceMotionChanged',
-        handleReduceMotionChanged as (event: AccessibilityEvent) => void,
-      )
-      AccessibilityInfo.removeEventListener(
-        'screenReaderChanged',
-        handleScreenReaderChanged as (event: AccessibilityEvent) => void,
-      )
+    const initializerKey = EVENT_NAME_TO_INITIALIZER[eventName]
+
+    if (!initializerKey) {
+      return
+    }
+
+    AccessibilityInfo[initializerKey]().then(setIsEnabled)
+    AccessibilityInfo.addEventListener(eventName, setIsEnabled)
+
+    return () => AccessibilityInfo.removeEventListener(eventName, setIsEnabled)
+  }, [eventName])
+
+  return isEnabled
+}
+
+export function useAccessibilityInfo(): {
+  screenReaderEnabled: Boolean
+  boldTextEnabled: Boolean
+  grayscaleEnabled?: Boolean
+  invertColorsEnabled?: Boolean
+  reduceMotionEnabled?: Boolean
+  reduceTransparencyEnabled?: Boolean
+} {
+  const screenReaderEnabled = useAccessibilityStateListener(
+    'screenReaderChanged',
+  )
+  const boldTextEnabled = useAccessibilityStateListener('boldTextChanged')
+
+  if (!SUPPORTS_RN60_ACCESSIBILITY_INFO_API) {
```

Comment:
Instead of explicitly checking for React Native 0.60, what do you think about letting `useAccessibilityStateListener` return `boolean | undefined`. That way the `useAccessibilityInfo` function doesn't need to do any branching (which will no longer violate eslint `react-hooks/rules-of-hooks`), and it will be easier to add support for new names in the future...

I also think it would be nice to pass in the event name to `useAccessibilityStateListener` instead of using a lookup table, since it's only used in one place and it currently just silently returns if there is a missmatch.

Lastly, I think it would be nice if we return `undefined` instead of `false` in the first render pass, when we don't yet know the value. That way the user could distinguish between it being actually disabled, and we just don't knowing the value.

Something like this:

```ts
function useAccessibilityStateListener(
  initializerName: keyof AccessibilityInfo,
  eventName: AccessibilityChangeEventName,
): boolean | undefined {
  const [isEnabled, setIsEnabled] = useState<boolean>()

  useEffect(() => {
    if (typeof AccessibilityInfo[initializerName] !== 'function') {
      return
    }

    AccessibilityInfo[initializerName]().then(setIsEnabled)
    AccessibilityInfo.addEventListener(eventName, setIsEnabled)

    return () => AccessibilityInfo.removeEventListener(eventName, setIsEnabled)
  }, [eventName])

  return isEnabled
}
```

```ts
export function useAccessibilityInfo(): {
  boldTextEnabled?: boolean
  grayscaleEnabled?: boolean
  invertColorsEnabled?: boolean
  reduceMotionEnabled?: boolean
  reduceTransparencyEnabled?: boolean
  screenReaderEnabled?: boolean
} {
  const boldTextEnabled = useAccessibilityStateListener(
    'isBoldTextEnabled',
    'boldTextChanged',
  )
  const grayscaleEnabled = useAccessibilityStateListener(
    'isGrayscaleEnabled',
    'grayscaleChanged',
  )
  const invertColorsEnabled = useAccessibilityStateListener(
    'isInvertColorsEnabled',
    'invertColorsChanged',
  )
  const reduceMotionEnabled = useAccessibilityStateListener(
    'isReduceMotionEnabled',
    'reduceMotionChanged',
  )
  const reduceTransparencyEnabled = useAccessibilityStateListener(
    'isReduceTransparencyEnabled',
    'reduceTransparencyChanged',
  )
  const screenReaderEnabled = useAccessibilityStateListener(
    'isScreenReaderEnabled',
    'screenReaderChanged',
  )

  return {
    boldTextEnabled,
    grayscaleEnabled,
    invertColorsEnabled,
    reduceMotionEnabled,
    reduceTransparencyEnabled,
    screenReaderEnabled,
  }
}
```

(I've also sorted the values alphabetically to help with git diffs when we add new values)

Reply:
Implemented the changes. I feel as though I've gone full circle on this haha. I think this is the  most concise.